fusillade/clouddirectory.py

```diff
@@ -0,0 +1,788 @@
+import functools
+import json
+import typing
+import boto3
+from fusillade.errors import FusilladeException
+from collections import namedtuple
+from enum import Enum, auto
+from urllib.parse import quote, unquote
+import os
+
+ad = boto3.client("clouddirectory")
+arn = "arn:aws:clouddirectory:us-east-1:861229788715:"
+
+
+def get_published_schema_from_directory(dir_arn: str) -> str:
+    schema = ad.list_applied_schema_arns(DirectoryArn=dir_arn)['SchemaArns'][0]
+    schema = schema.split('/')[-2:]
+    schema = '/'.join(schema)
+    return f"{arn}schema/published/{schema}"
+
+
+def cleanup_directory(dir_arn: str):
+    ad.disable_directory(DirectoryArn=dir_arn)
+    ad.delete_directory(DirectoryArn=dir_arn)
+
+
+def cleanup_schema(sch_arn: str):
+    ad.delete_schema(SchemaArn=sch_arn)
+
+
+def publish_schema(name: str, version: str):
+    # don't create if already created
+    try:
+        dev_schema_arn = ad.create_schema(Name=name)['SchemaArn']
+    except ad.exceptions.SchemaAlreadyExistsException:
+        dev_schema_arn = f"{arn}schema/development/{name}"
+
+    # update the schema
+    pkg_root = os.path.abspath(os.path.join(os.path.dirname(__file__), '..'))  # noqa
```

Comment:
I'd put `pkg_root = os.path.abspath(os.path.join(os.path.dirname(__file__), '..'))  # noqa` at the top of the file.  Useful var to have.